agir/events/views/event_views.py

```diff
@@ -71,17 +81,184 @@
 ]
 
 
-class EventListView(SearchByZipcodeBaseView):
+def get_date(date_str):
+    try:
+        date = datetime.strptime(date_str, "%d/%m/%Y").date()
+    except ValueError:
+        return None
+    return date
+
+
+def get_location(addresse):
+    if not addresse:
+        return None
+
+    results = get_results_from_ban({"q": addresse, "limit": 5})
+
+    if not results:
+        return None
+
+    for feature in results["features"]:
+        if feature["geometry"]["type"] == "Point":
+            return Point(*feature["geometry"]["coordinates"], srid=4326)
+
+    return None
+
+
+class EventListView(SearchByZipcodeBaseView, FormView):
     """List of events, filter by zipcode
     """
 
     template_name = "events/event_list.html"
     context_object_name = "events"
     form_class = SearchEventForm
+    success_url = reverse_lazy("list_all_events")
+    event_founded = []
+    events_by_page = 10
+
+    is_next_page = None
+    is_previous_page = None
+    page_nb = 0
+    is_next_page_more = None
+
+    def get(self, request, *args, **kwargs):
+        url_args = request.GET
+        text_query = url_args.get("text_query", None)
+        date_begin = get_date(url_args.get("date_begin", ""))
+        date_end = get_date(url_args.get("date_end", ""))
+        location = (
+            get_location(url_args.get("address", "")) or request.user.person.coordinates
+        )
+        distance_max = float(url_args.get("distance_max", "0"))
+        self.page_nb = int(url_args.get("page_nb", "0"))
+        self.event_founded = self.event_full_text_search(
+            text_query, date_begin, date_end, location, distance_max, self.page_nb
+        )
+
+        return super().get(request, *args, **kwargs)
 
     def get_base_queryset(self):
         return Event.objects.upcoming().order_by("start_time")
 
+    def get_form_kwargs(self):
+        return {
+            "data": {
+                "date_begin": self.request.GET.get("date_begin", ""),
+                "date_end": self.request.GET.get("date_end", ""),
+                "text_query": self.request.GET.get("text_query", ""),
+                "address": self.request.GET.get("address", ""),
+                "distance_max": self.request.GET.get("distance_max", "inf"),
+                "page_nb": self.request.GET.get("page_nb", "0"),
+            }
+        }
+
+    def event_full_text_search(
+        self, text_query, date_begin, date_end, location, distance_max, page_number
+    ):
+        q = PrefixSearchQuery(text_query, config="fr")
+        qs = Event.objects.annotate(rank=SearchRank(F("search"), q))
+        qs = qs.filter(rank__gt=0.1).order_by("-rank")
```

Comment:
Pourquoi ne pas filtrer `qs.filter(q)` ? Du coup tu attrapes tous les résultats avec un terme qui matchent, y compris ceux avec un rang potentiellement plus faible que `0.1` (ce qui n'est pas dramatique dans le cas général, puisque tu ordonnes par `rank`.

Reply:
`qs.filter(q)` lève une erreur : `'PrefixSearchQuery' object is not iterable` mais j'immagine que `qs.filter(rank__gt=0)` fait le taf.